test/e2e/scale_test.go

```diff
@@ -0,0 +1,151 @@
+				InfrastructureProvider: ptr.To(clusterctl.DefaultInfrastructureProvider),
+				BootstrapClusterProxy:  bootstrapClusterProxy,
+				ArtifactFolder:         artifactFolder,
+				Flavor:                 ptr.To(testSpecificSettingsGetter().FlavorForMode("topology-runtimesdk")),
+				SkipUpgrade:            true,
+				SkipCleanup:            skipCleanup,
+				ClusterClassName:       getVariableOrFallback(testSpecificSettingsGetter().Variables["CLUSTER_CLASS_NAME"], e2eConfig.GetVariable("CLUSTER_CLASS_NAME")),
+
+				// ClusterCount can be overwritten via `CAPI_SCALE_CLUSTER_COUNT`.
+				ClusterCount: ptr.To[int64](5),
+				// Concurrency can be overwritten via `CAPI_SCALE_CONCURRENCY`.
+				Concurrency: ptr.To[int64](20),
+				// ControlPlaneMachineCount can be overwritten via `CAPI_SCALE_CONTROL_PLANE_MACHINE_COUNT`.
+				ControlPlaneMachineCount: ptr.To[int64](3),
+				// WorkerPerMachineDeploymentCount can be overwritten via `CAPI_SCALE_WORKER_PER_MACHINE_DEPLOYMENT_COUNT`.
+				WorkerPerMachineDeploymentCount: ptr.To[int64](1),
+				// MachineDeploymentCount can be overwritten via `CAPI_SCALE_MACHINE_DEPLOYMENT_COUNT`.
+				MachineDeploymentCount: ptr.To[int64](3),
+				// AdditionalClusterClassCount can be overwritten via `CAPI_SCALE_ADDITIONAL_CLUSTER_CLASS_COUNT`.
+				AdditionalClusterClassCount: ptr.To[int64](4),
+				// DeployClusterInSeparateNamespaces can be overwritten via `CAPI_SCALE_DEPLOY_CLUSTER_IN_SEPARATE_NAMESPACES`.
+				DeployClusterInSeparateNamespaces: ptr.To(true),
+				// UseCrossNamespaceClusterClass can be overwritten via `CAPI_SCALE_USE_CROSS_NAMESPACE_CLUSTER_CLASS`.
+				UseCrossNamespaceClusterClass: ptr.To(false),
+
+				// The runtime extension gets deployed to the test-extension-system namespace and is exposed
+				// by the test-extension-webhook-service.
+				// The below values are used when creating the cluster-wide ExtensionConfig to refer
+				// the actual service.
+				ExtensionServiceNamespace: "capv-test-extension",
+				ExtensionServiceName:      "capv-test-extension-webhook-service",
+				ExtensionConfigName:       "capv-scale-test-extension",
+
+				PostNamespaceCreated: func(managementClusterProxy framework.ClusterProxy, workloadClusterNamespace string) {
+					testSpecificSettingsGetter().PostNamespaceCreatedFunc(managementClusterProxy, workloadClusterNamespace)
+
+					// Overwrite some variables with placeholders which get replaced later on in PostScaleClusterNamespaceCreated
+					testSpecificVariables := map[string]string{
+						"VSPHERE_SERVER":              scaleClusterVSphereServerPlaceholder,
+						"VSPHERE_TLS_THUMBPRINT":      scaleClusterVSphereTLSThumbprintPlaceholder,
+						"CONTROL_PLANE_ENDPOINT_IP":   scaleClusterControlPlaneEndpointIPPlaceholder,
+						"CONTROL_PLANE_ENDPOINT_PORT": scaleClusterControlPlaneEndpointPortPlaceholder,
+						// Cluster API uses a hard coded name for the ExtensionConfig.
+						"RUNTIME_SDK_EXTENSIONCONFIG_NAME": "capv-scale-test-extension",
```

Comment:
I assume we could drop this if we would use EXTENSION_CONFIG_NAME in our YAMLs

Reply:
Yes, but then it could collide with a parallel runtime sdk test (overlapping extensonconfig name)

Reply:
I think I don't get it. Doesn't the core CAPI ExtensionConfigName input field allow to use a different ExtensionConfig in each test / namespace?

Reply:
ah, now I got it.